youtube_comedy_slam/youtube_comments_api.py

```diff
@@ -0,0 +1,153 @@
+        videoId=video_id
+    ).execute()
+
+
+    # pp = pprint.PrettyPrinter(indent=4)
+    # pp.pprint(results)
+
+    write_header = not os.path.isfile(csvfile)
+
+    ## Setup csv file
+    fieldnames = ['videoId', 'textDisplay', 'canReply', 'totalReplyCount', 'kind', 'etag', 'id', 'authorChannelId',
+                  'authorDisplayName', 'likeCount', 'publishedAt', 'updatedAt', 'viewerRating']
+    with open(csvfile, 'a') as fcsv:
+        writer = csv.DictWriter(fcsv, fieldnames=fieldnames)
+        if write_header:
+            writer.writeheader()
+
+        def get_comment_page(results, ignoreFirst = False):
+            first = True
+            for item in results["items"]:
+                if first:
+                    first = False
+                    continue
+                snippet = item["snippet"]
+                comment = snippet["topLevelComment"]["snippet"]
+                videoId = comment["videoId"]
+                textDisplay = comment["textDisplay"]
+                canReply = snippet["canReply"]
+                totalReplyCount = snippet["totalReplyCount"]
+                kind = item["kind"]
+                etag = snippet["topLevelComment"]["etag"]
+                id = snippet["topLevelComment"]["id"]
+                authorChannelId = comment["authorChannelId"] if "authorChannelId" in comment else ""
+                authorDisplayName = comment["authorDisplayName"]
+                likeCount = comment["likeCount"]
+                publishedAt = comment["publishedAt"]
+                updatedAt = comment["updatedAt"]
+                viewerRating = comment["viewerRating"]
+                writer.writerow(
+                    {'viewerRating': viewerRating, 'videoId': videoId, 'textDisplay': textDisplay, 'canReply': canReply,
+                     'totalReplyCount': totalReplyCount, 'kind': kind, 'etag': etag, 'id': id,
+                     'authorChannelId': authorChannelId, 'authorDisplayName': authorDisplayName, 'likeCount': likeCount,
+                     'publishedAt': publishedAt, 'updatedAt': updatedAt, 'viewerRating': viewerRating, 'videoId': videoId})
+
+        get_comment_page(results)
+
+        counter = 0
+        while ('nextPageToken' in results) and (counter < 3):
+            # Fetch next page
+            nextPageToken = results['nextPageToken']
+            results = youtube.commentThreads().list(
+                part="snippet,replies",
+                textFormat="plainText",
+                maxResults=100,
+                pageToken = nextPageToken,
+                videoId=video_id
+            ).execute()
+            # You have to ignore the first result
+            get_comment_page(results, ignoreFirst=True)
+            counter += 1
+
+    return results
+
+
+def get_comments(youtube, parent_id):
+    results = youtube.comments().list(
+        part="snippet",
+        parentId=parent_id,
+        textFormat="plainText"
+    ).execute()
+
+    print(results)
```

Comment:
In python usually `logging` is used instead of `print` (prints are being mostly used only for debugging)